src/bundler/bundler_web_server.rs

```diff
@@ -0,0 +1,204 @@
+use crate::backends::storage_backend::StorageBackend;
+use anyhow::Result;
+use aruna_file::{
+    streamreadwrite::ArunaStreamReadWriter,
+    transformer::{FileContext, ReadWriter},
+    transformers::{
+        async_sender_sink::AsyncSenderSink, decrypt::ChaCha20Dec, gzip_comp::GzipEnc, tar::TarEnc,
+        zstd_decomp::ZstdDec,
+    },
+};
+use axum::{
+    body::StreamBody,
+    error_handling::HandleErrorLayer,
+    extract::{Path, State},
+    response::{IntoResponse, Response},
+    routing::get,
+    BoxError, Router,
+};
+use hyper::StatusCode;
+use std::{sync::Arc, time::Duration};
+use tokio::sync::Mutex;
+use tower::{buffer::BufferLayer, limit::RateLimitLayer, ServiceBuilder};
+
+use super::bundler::Bundler;
+
+struct RequestError(anyhow::Error);
+
+pub async fn run_axum(
+    bundler_addr: String,
+    bundler: Arc<Mutex<Bundler>>,
+    backend: Arc<Box<dyn StorageBackend>>,
+) -> Result<()> {
+    // build our application with a route
+    let app = Router::new()
+        // `GET /` goes to `root`
+        .route("/:bundle_id/:file_name", get(get_bundle))
+        .with_state((bundler.clone(), backend))
+        .layer(
+            ServiceBuilder::new()
+                .layer(HandleErrorLayer::new(|err: BoxError| async move {
+                    (
+                        StatusCode::INTERNAL_SERVER_ERROR,
+                        format!("Unhandled error: {}", err),
+                    )
+                }))
+                .layer(BufferLayer::new(1024))
+                .layer(RateLimitLayer::new(5, Duration::from_secs(1))),
+        );
+
+    // Axum Server
+    // run our app with hyper, listening globally on port 3000
+    axum::Server::bind(&bundler_addr.parse()?)
+        .serve(app.into_make_service())
+        .await?;
+
+    Ok(())
+}
+
+async fn get_bundle(
+    State((bundler, backend)): State<(Arc<Mutex<Bundler>>, Arc<Box<dyn StorageBackend>>)>,
+    Path((bundle_id, file_name)): Path<(String, String)>,
+) -> Result<impl IntoResponse, RequestError> {
+    if !file_name.ends_with(".tar.gz") {
+        return Err(RequestError(anyhow::anyhow!(
+            "Only .tar.gz is currently supported"
+        )));
+    }
+    let bundler_store = &bundler.lock().await.bundle_store;
+    let bundles = bundler_store.get(&bundle_id);
+    let (tx, rx) = async_channel::bounded(10);
+    let (file_info_sender, file_info_receiver) = async_channel::bounded(10);
+    let file_info_recv_clone = file_info_receiver.clone();
+    let file_info_sender_clone = file_info_sender.clone();
+    let tx_clone = tx.clone();
+    let rx_clone = rx.clone();
+    match bundles {
+        Some(b) => match b {
+            Some(bundle) => {
+                // TODO: check if bundle is expired
```

Comment:
Is this ToDo for the future, was it simply forgotten during implementation, or is it no longer relevant?

Reply:
Currently we expect this to be handled on the server-side, but for the future if this gets more independent we should check the expiry dataproxy sided as well -> So long we could leave the comment just to reminder that the work there is not completely done